modules/azurerm_vm/variables.tf

```diff
@@ -1,115 +1,149 @@
+  type        = string
+  default     = "latest"
 }
 
-variable name {
-  type = string
+variable "name" {
+  description = "VM name"
+  type        = string
 }
 
-variable data_disk_count {
-  default = 0
+variable "data_disk_count" {
+  description = "Number of disks to be created"
+  type        = number
+  default     = 0
 }
 
-variable data_disk_size {
-  default = 128
+variable "data_disk_size" {
+  description = "Specifies the size of the managed disk to create in gigabytes."
+  type        = number
+  default     = 128
 }
 
-variable data_disk_caching {
-  default = "ReadWrite"
+variable "data_disk_caching" {
+  description = "Specifies the caching requirements for this Data Disk. Possible values include None, ReadOnly and ReadWrite"
+  type        = string
+  default     = "ReadWrite"
 }
 
-variable data_disk_storage_account_type {
-  default     = "Standard_LRS"
+variable "data_disk_storage_account_type" {
   description = "The type of storage to use for the managed disk. Possible values are Standard_LRS, Premium_LRS, StandardSSD_LRS or UltraSSD_LRS."
+  type        = string
+  default     = "Standard_LRS"
 }
 
 variable "data_disk_zone" {
   description = "Specifies the Availability Zone in which this Managed Disk should be located. Changing this property forces a new resource to be created."
+  type        = string
   default     = null
 }
 
-variable os_disk_size {
-  default = 64
+variable "os_disk_size" {
+  description = "The Size of the Internal OS Disk in GB, if you wish to vary from the size used in the image this Virtual Machine is sourced from."
+  type        = number
+  default     = 64
 }
 
-variable os_disk_storage_account_type {
-  default     = "Standard_LRS"
+variable "os_disk_storage_account_type" {
   description = "The Type of Storage Account which should back this the Internal OS Disk. Possible values are Standard_LRS, StandardSSD_LRS and Premium_LRS. Changing this forces a new resource to be created"
+  type        = string
+  default     = "Standard_LRS"
 }
 
-variable os_disk_caching {
-  default = "ReadOnly"
-}
-
-variable enable_accelerated_networking {
-  default = false
+variable "os_disk_caching" {
+  description = "The Type of Caching which should be used for the Internal OS Disk. Possible values are None, ReadOnly and ReadWrite."
+  type        = string
+  default     = "ReadOnly"
 }
 
-variable create_vm {
-  default = false
+variable "enable_accelerated_networking" {
+  description = "Enables network accelaration for VMs. By default enabled for the nfs and jump VMs. For any other VM the default is false"
+  type        = bool
+  default     = false
 }
 
-variable cloud_init {
-  default = ""
+variable "cloud_init" {
+  description = "cloud_init config"
+  type        = string
+  default     = ""
 }
 
-variable create_public_ip {
-  default = false
+variable "create_public_ip" {
+  description = "Reference to a Public IP Address to associate with this NIC"
```

Comment:
does this boolean also create a public IP address or just reference one that already exists, not sure given the text here?

Reply:
Updated the description in the latest commit.